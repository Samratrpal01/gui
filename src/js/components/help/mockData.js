export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
