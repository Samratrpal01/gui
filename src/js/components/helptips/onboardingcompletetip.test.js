import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import OnboardingCompleteTip from './onboardingcompletetip';
import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';

const mockStore = configureStore([thunk]);

describe('OnboardingCompleteTip Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      ...defaultState,
      app: {
        ...defaultState.app,
        features: {
          ...defaultState.app.features,
          hasMultitenancy: true,
          isHosted: true
        }
      }
    });
    jest.spyOn(global, 'encodeURIComponent').mockImplementationOnce(() => 'http%3A%2F%2Ftest.com');
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <Provider store={store}>
        <OnboardingCompleteTip targetUrl="https://test.com" />
      </Provider>
    );
    const view = baseElement;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
