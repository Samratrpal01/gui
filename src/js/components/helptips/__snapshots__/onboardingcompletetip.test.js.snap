// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingCompleteTip Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-1 {
  opacity: 0.9;
}

.emotion-2 {
  z-index: 1500;
  pointer-events: auto;
}

.emotion-3 {
  background-color: #1976d2;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  color: #9e9e9e;
  font-size: 14px;
  max-width: 350px;
  padding: 12px 18px;
  width: 350px;
}

.emotion-3 a {
  color: #9e9e9e;
}

.emotion-3.MuiTooltip-tooltipPlacementTop {
  margin-left: 30px;
  margin-bottom: 0;
  margin-top: calc(30 + 12px);
}

.emotion-3.MuiTooltip-tooltipPlacementRight {
  margin-top: 15px;
}

.emotion-3.MuiTooltip-tooltipPlacementBottom {
  margin-left: 30px;
}

.emotion-3.MuiTooltip-tooltipPlacementLeft {
  margin-top: 15px;
}

.emotion-4 {
  background-color: rgba(97, 97, 97, 0.92);
  border-radius: 4px;
  color: #fff;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  padding: 4px 8px;
  font-size: 0.6875rem;
  max-width: 300px;
  margin: 2px;
  word-wrap: break-word;
  font-weight: 500;
}

.MuiTooltip-popper[data-popper-placement*="left"] .emotion-4 {
  transform-origin: right center;
  margin-right: 14px;
}

.MuiTooltip-popper[data-popper-placement*="right"] .emotion-4 {
  transform-origin: left center;
  margin-left: 14px;
}

.MuiTooltip-popper[data-popper-placement*="top"] .emotion-4 {
  transform-origin: center bottom;
  margin-bottom: 14px;
}

.MuiTooltip-popper[data-popper-placement*="bottom"] .emotion-4 {
  transform-origin: center top;
  margin-top: 14px;
}

.emotion-5 {
  background-color: #fff;
}

.emotion-5:hover {
  background-color: #fff;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #9c27b0;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #7b1fa2;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-9:hover {
    background-color: #9c27b0;
  }
}

.emotion-9:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-9.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-9.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

<body>
  <div>
    <div
      aria-labelledby="onboarding-finished"
      class="tooltip onboard-icon onboard-tip"
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
  </div>
  <div
    class="emotion-1 MuiTooltip-popper MuiTooltip-popperInteractive emotion-2 MuiPopperUnstyled-root"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    id="onboarding-finished"
    role="tooltip"
    style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="emotion-3 MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom emotion-4"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="content"
      >
        <p>
          Great work! You updated your device with the new Release!
        </p>
        <div
          class="margin-bottom-small margin-top-small"
        >
          Your device is now running the updated version of the software. At
          <div
            class="flexbox centered"
            style="margin: 5px 0px;"
          >
            <a
              class="emotion-5 MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root emotion-5 button emotion-7"
              href="http://192.168.10.141:85/index.html?source=http%3A%2F%2Ftest.com"
              tabindex="0"
              target="_blank"
            >
              Go to http://192.168.10.141:85
              <span
                class="MuiTouchRipple-root emotion-8"
              />
            </a>
          </div>
          you should now see "Hello world" in place of the webpage you saw previously. If you continue to see the webpage you saw previously you might have to refresh the page.
        </div>
        <p>
          You've now got a good foundation in how to use Mender. Look for more help hints in the UI as you go along.
        </p>
        What next?
        <div>
          Proceed to one of the following tutorials (listed in recommended order):
          <ol>
            <li>
              <a
                href="https://docs.mender.io/get-started/deploy-a-system-update"
                rel="noopener noreferrer"
                target="_blank"
              >
                Deploy a system update
              </a>
            </li>
            <li>
              <a
                href="https://docs.mender.io/get-started/deploy-a-container-update"
                rel="noopener noreferrer"
                target="_blank"
              >
                Deploy a container update
              </a>
            </li>
          </ol>
        </div>
        <div
          class="flexbox"
        >
          <div
            style="flex-grow: 1;"
          />
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  emotion-9"
            tabindex="0"
            type="button"
          >
            Close
            <span
              class="MuiTouchRipple-root emotion-8"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
